keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1120 @@
+    
+    // Dependencies is a list of other Applications that this Application depends on.
+    Dependencies []sting `json:"dependencies" protobuf:"bytes,3,opt,name=dependencies"`
+    
+    // selector is a label query over kinds that created by the application. It must match the component objects' labels.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    Selector *metav1.LabelSelector `json:"selector" protobuf:"bytes,4,opt,name=selector"`
+    
+    // HealthCheck is an optional application level health check.
+    HealthCheck* ApplicationHealthCheck `json:"healthCheck" protobuf:"bytes,5,opt,name=healthCheck"`
+    
+    // Version is an optional version indicator for the application.
+    Version string `json:"version,omitempty" protobuf:"bytes,6,opt,name=version"`
+    
+    // AboutURL is a URL pointing to any additional information about the application.
+    AboutURL string `json:"url,omitempty" protobuf:"bytes,7,opt,name=url"`
+    
+    // Description is a brief string description of the application.
+    Description string `json:"description,omitempty" protobuf:"bytes,8,opt,name=description"`
+    
+    // Maintainers is an optional list of maintainers of the application
+    Maintainers []string `json:"maintainers,omitempty" protobuf:"bytes,9,opt,name=maintainers"`
+    
+    // Keywords is an optional list of key words associated with the application (e.g. MySQL, RDBMS, database).
+    Keywords []string `json:"keywords,omitempty" protobuf:"bytes,10,opt,name=keywords"`
+    
+    // LicenseURL is a an optional URL pointing to any licensing information for the application.
+    LicenseURL string `json:"license,omitempty" protobuf:"bytes,11,opt,name=license"`
+    
+    // DashboardURL is a an optional URL pointing to the applications dashboard.
+    DashboardURL string `json:"dashboard,omitempty" protobuf:"bytes,12,opt,name=dashboard"`
+	
+}
+
+// The Application object acts as an aggregator for components that comprise an application. Its Spec.Components 
+// indicate the components that comprise the application and the GroupVersionKinds of the those components. 
+// Its SpecSSelector is used to list and watch the Components that correspond to a GVK. All components of an 
+// Application should be labeled such the Application's Selector matches. An Applications Spec.Dependencies indicate 
+// the other Applications on which the Application depends.
+type Application struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+    
+    // Spec contains the specification for the Application
+    Spec ApplicationSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+    
+    // Status contains the Status of the application with respect to the installed components.
+    Status ApplicationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
I'm really confused about what this is. The spec makes it sound like status is tracked for each object and dependency, but I can't see how this is done with just one single ApplicationStatus object that doesn't seem to separate out the individual components of this application, and instead just has a []string of Installed. I can't see how I, as an operator, would be able to reconcile the status with the actual state of things.